pkg/encryption/aesgcm.go

```diff
@@ -36,14 +35,14 @@ type aesgcmEncrypter struct {
 func NewAESGCMEncrypter(key *storj.Key, startingNonce *AESGCMNonce, encryptedBlockSize int) (Transformer, error) {
 	block, err := aes.NewCipher(key[:])
 	if err != nil {
-		return nil, err
+		return nil, Error.Wrap(err)
 	}
 	aesgcmEncrypt, err := cipher.NewGCM(block)
 	if err != nil {
-		return nil, err
+		return nil, Error.Wrap(err)
 	}
 	if encryptedBlockSize <= aesgcmEncrypt.Overhead() {
-		return nil, Error.New("block size too small")
+		return nil, EncryptedBlockSizeTooSmallError.New(strconv.Itoa(encryptedBlockSize))
```

Comment:
This sounds like a Java error :)

Reply:
Any suggestion how to improve it?

Reply:
`ErrInvalidParams.New("encrypted block size %d too small", encryptedBlockSize)`. We only need specific error class when we need to distinguish and take action on the specific situation.

Reply:
I pushed another commit? How is it now?

I intentionally left `ErrDecryptFailed` as a separate class, so I can have a more user-friendly error message when decryption fails. This was the main motivation for this PR.

Reply:
I was making a joke but I like the new error type better.